src/nupic/encoders/CoordinateEncoder.cpp

```diff
@@ -0,0 +1,128 @@
+/* ---------------------------------------------------------------------
+ * Numenta Platform for Intelligent Computing (NuPIC)
+ * Copyright (C) 2019, David McDougall
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses.
+ *
+ * http://numenta.org/licenses/
+ * --------------------------------------------------------------------- */
+
+/** @file
+ * Implement the CoordinateEncoder
+ */
+
+#define _USE_MATH_DEFINES // Define before including anything
+#include <nupic/encoders/CoordinateEncoder.hpp>
+#include <math.h> // tgamma, M_PI
+#include <nupic/utils/MurmurHash3.hpp>
+
+using namespace std;
+using namespace nupic::encoders;
+
+CoordinateEncoder::CoordinateEncoder(const CoordinateEncoderParameters &parameters)
+  { initialize( parameters ); }
+
+void CoordinateEncoder::initialize( const CoordinateEncoderParameters &parameters ) {
+  // Check parameters:  This has all of the same constraints as the RDSE does.
+  // The RDSE will raise an exception if the parameters are bad.
+  RDSE rdse( parameters );
+  NTA_CHECK( parameters.numDimensions > 0u );
+
+  BaseEncoder<const vector<Real64> &>::initialize({ parameters.size });
+  args_ = parameters;
+  // Fill in remaining parameters.
+  args_.activeBits = rdse.parameters.activeBits;
+  args_.sparsity   = rdse.parameters.sparsity;
+  args_.seed       = rdse.parameters.seed;
+  // args_.resolution = TODO
```

Comment:
resolution TODO

Reply:
I think actually I'm going to remove the `resolution` option.  It can be added back in later without changing API thanks to parameter structure.